posthog/temporal/data_imports/sources/revenuecat/revenuecat.py

```diff
@@ -0,0 +1,244 @@
+
+    if response.status_code == 429:
+        logger.warning(f"RevenueCat: rate limited (429), will retry...")
+        raise RevenueCatRetryableError(f"Rate limited: {response.status_code} - {response.text}")
+
+    if response.status_code >= 500:
+        logger.warning(f"RevenueCat: server error ({response.status_code}), will retry...")
+        raise RevenueCatRetryableError(f"Server error: {response.status_code} - {response.text}")
+
+    response.raise_for_status()
+
+    data = response.json()
+    return data.get("items", []), data.get("next_page")
+
+
+def _paginate(
+    url: str,
+    headers: dict[str, str],
+    logger: FilteringBoundLogger,
+    starting_after: str | None = None,
+) -> Iterator[dict]:
+    params: dict[str, str | int] = {"limit": DEFAULT_LIMIT}
+
+    if starting_after:
+        params["starting_after"] = starting_after
+        logger.debug(f"RevenueCat: using incremental sync, starting_after={starting_after}")
+
+    while True:
+        items, next_page = _fetch_page(url, headers, params, logger)
+
+        yield from items
+
+        if not next_page:
+            break
+
+        url = next_page
+        params = {}
+
+
+def get_rows(
+    api_key: str,
+    project_id: str,
+    endpoint: str,
+    logger: FilteringBoundLogger,
+    should_use_incremental_field: bool = False,
+    db_incremental_field_last_value: str | None = None,
+) -> Iterator[dict]:
+    headers = {
+        "Authorization": f"Bearer {api_key}",
+        "Content-Type": "application/json",
+        "User-Agent": "PostHog-DataPipelines",
+    }
+
+    resources: dict[str, RevenueCatResource | RevenueCatNestedResource] = {
+        "Apps": RevenueCatResource(path=f"/projects/{project_id}/apps"),
+        "CustomerActiveEntitlements": RevenueCatNestedResource(
+            path=f"/projects/{project_id}/customers/{{customer_id}}/active_entitlements",
+            parent_path=f"/projects/{project_id}/customers",
+            parent_id_field="id",
+            nested_parent_param="customer_id",
+        ),
+        "CustomerAliases": RevenueCatNestedResource(
+            path=f"/projects/{project_id}/customers/{{customer_id}}/aliases",
+            parent_path=f"/projects/{project_id}/customers",
+            parent_id_field="id",
+            nested_parent_param="customer_id",
+        ),
+        "CustomerPurchases": RevenueCatNestedResource(
+            path=f"/projects/{project_id}/customers/{{customer_id}}/purchases",
+            parent_path=f"/projects/{project_id}/customers",
+            parent_id_field="id",
+            nested_parent_param="customer_id",
+        ),
+        "CustomerSubscriptions": RevenueCatNestedResource(
+            path=f"/projects/{project_id}/customers/{{customer_id}}/subscriptions",
+            parent_path=f"/projects/{project_id}/customers",
+            parent_id_field="id",
+            nested_parent_param="customer_id",
+        ),
```

Comment:
Oh one more question, can these not be incremental based on the parent `customer_id`? I think we do this for some nested stripe resources too